### PR TITLE
[v7r1] Allows the transition to JSON serialization to start

### PR DESCRIPTION
**Pull request by chaen**
This PR introduces the necessary code and tools to do the transition to JSON serialization. 
Basically, instead of doing 4 releases (i.e. transition of about 40 months), we control the changes with environment variable. Of course, this needs that all the JSON tasks that are still opened are addressed.

BEGINRELEASENOTES
*Core
NEW: Allows for json serialization transition

ENDRELEASENOTES

### CODE REVIEW

**Comment by atsareg** [comment]
As far as I can see, by default the behavior is not changed. Is it correct ? I will then merge it into the final release

**Comment by fstagni** [comment]
The behavior should not change with code, yet, but controlled by the environment variables as Chris did.

**Comment by chaen** [comment]
yes that's correct. And the unit tests allowed to show this nasty corner case with `None` :-)

File: docs/source/AdministratorGuide/technologyPreviews.rst [hunk 1]
```diff
@@ -22,3 +22,21 @@ M2Crypto (or any standard tool that respects TLS..) will be stricter than PyGSI.
...
 * ComponentInstaller screwed: like any change you do on your hosts, the ComponentInstaller will duplicate the entry. So if you change the CS to put FQDN, the machine will appear twice. 
+
+
+.. _jsonSerialization:
+
+JSON Serialization
+==================
+
+We aim at replacing the DEncode serialization over the network with json serialization. In order to be smooth, the transition needs to happen in several steps:
+* DISET -> DISET: as it is now. We encode DISET, and decode DISET
```
Comment (andresailer) [inline docs/source/AdministratorGuide/technologyPreviews.rst:33]:
```suggestion

* DISET -> DISET: as it is now. We encode DISET, and decode DISET
```
Need an empty line before a bullet list

File: docs/source/AdministratorGuide/technologyPreviews.rst [hunk 2]
```diff
@@ -22,3 +22,21 @@ M2Crypto (or any standard tool that respects TLS..) will be stricter than PyGSI.
...
+* DISET -> DISET, JSON: we send DISET, try to decode with DISET, fallback to JSON if it fails. This step is to make sure all clients will be ready to understand JSON whenever it comes.
+* JSON -> DISET, JSON: we send JSON, attempt to decode with DISET, fallback to JSON if it fails. This step is to make sure that if we still have some old clients lying around, we are still able to understand them.
+* JSON -> JSON: final step, goodbye DISET.
+
+The changes from one stage to the next is controlled by environment variables, and it can go to your own pace:
+* `DIRAC_USE_JSON_DECODE`: must be the first one. Enables the DISET,JSON decoding
```
Comment (andresailer) [inline docs/source/AdministratorGuide/technologyPreviews.rst:39]:
```suggestion

* ``DIRAC_USE_JSON_DECODE``: must be the first one. Enables the DISET,JSON decoding
```

Comment (atsareg) [reply docs/source/AdministratorGuide/technologyPreviews.rst:39]: I will add this changes when tagging. Any other suggestions ?

Comment (andresailer) [reply docs/source/AdministratorGuide/technologyPreviews.rst:39]: Thanks! 
Nothing else at the moment

File: docs/source/AdministratorGuide/technologyPreviews.rst (hunk 1 above)
Comment (chaen) [reply docs/source/AdministratorGuide/technologyPreviews.rst:33]: ARGH ! I thought this was only for verbatim environment :'(
